app/components/render-zybook/component.js

```diff
@@ -0,0 +1,51 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+
+    /*
+    @property zybook
+    @type {Object}
+    @default null
+     */
+    zybook: null,
+    classNames: [ 'render-zybook' ],
+
+    /*
+    @event term_description
+    @param {}
+    @return {zybook.academic_term.name, zybook.academic_term.year}
+     */
+    term_description: function() {
+        if (this.get('zybook.academic_term.name') === 'None') {
+            return '';
+        }
+        return `${this.get('zybook.academic_term.name')} ${this.get('zybook.academic_term.year')}`;
+    }.property('zybook.academic_term.name', 'zybook.academic_term.year'),
+
+    /*
+     @event subject_description
+     @param {}
+     @return {zybook.subjects.@each.name}
+     */
+
+    subject_description: function() {
+        const subs = [];
+
+        this.get('zybook.subjects').forEach(sub => {
```

Comment:
Can use the `map` method for arrays here. Ex: `subs = this.get('zybook.subjects').map( ... ) `.